modules/openapi-generator/src/main/resources/python/python-experimental/api_client.mustache

```diff
@@ -542,3 +561,180 @@ class ApiClient(object):
+
+        # Get current time and generate RFC 1123 (HTTP/1.1) date/time string.
+        from email.utils import formatdate
+        now = datetime.datetime.now()
+        stamp = mktime(now.timetuple())
+        cdate=formatdate(timeval=stamp, localtime=False, usegmt=True)
+        created=now.strftime("%s")
+
+        signed_headers = {}
+        signed_header_dict = {}
+        for hdr_key in self.configuration.signed_headers:
+            hdr_key = hdr_key.lower()
+            if hdr_key == '(request-target)':
+                value = request_target
+            elif hdr_key == '(created)':
+                value = created
+            elif hdr_key == 'date':
+                value = cdate
+                signed_header_dict['Date'] = '{0}'.format(cdate)
+            elif hdr_key == 'digest':
+                request_body = body.encode()
+                body_digest, digest_prefix = self.get_message_digest(request_body)
+                b64_body_digest = b64encode(body_digest.digest())
+                value = digest_prefix + b64_body_digest.decode('ascii')
+                signed_header_dict['Digest'] = '{0}{1}'.format(digest_prefix, b64_body_digest.decode('ascii'))
+            elif hdr_key == 'host':
+                value = target_host
+                signed_header_dict['Host'] = '{0}'.format(target_host)
+            else:
+                value = headers[hdr_key]
+            signed_headers[hdr_key] = value
+
+        if len(self.configuration.signed_headers) == 0:
+            signed_headers['(created)'] = created
+
+        string_to_sign = self.get_str_to_sign(signed_headers)
+
+        digest, digest_prefix = self.get_message_digest(string_to_sign.encode())
+        b64_signed_msg = self.sign_digest(digest)
+
+        auth_header = self.get_authorization_header(signed_headers, b64_signed_msg, created)
+        signed_header_dict['Authorization'] = '{0}'.format(auth_header)
+
+        return signed_header_dict
+
+    def get_message_digest(self, data):
+        """
+        Calculates and returns a cryptographic digest of a specified HTTP request.
+
+        :param data: The message to be hashed with a cryptographic hash.
+        :return: The message digest encoded as a byte string. 
+        """
+        if self.configuration.signing_scheme in ["rsa-sha512", "hs2019"]:
+            digest = SHA512.new()
+            prefix = "SHA-512="
+        elif self.configuration.signing_scheme in ["rsa-sha256"]:
+            digest = SHA256.new()
+            prefix = "SHA-256="
+        else:
+            raise Exception(
+                "Unsupported signing algorithm: {0}".format(self.configuration.signing_scheme))
+        digest.update(data)
+        return digest, prefix
+
+    def sign_digest(self, digest):
+        """
+        Signs a message digest with a private key specified in the configuration.
+
+        :param digest: digest of the HTTP message.
+        :return: the HTTP message signature encoded as a byte string.
```

Comment:
Can you add type descriptions to these params?
b64encode(signature) does not return type bytes in python2.7, it returns type string

Reply:
doing.

Reply:
updated description. It is actually a hashing object.